examples/cluster/plot_adjusted_for_chance_measures.py

```diff
@@ -3,135 +3,217 @@
 
-    for i, k in enumerate(n_clusters_range):
-        for j in range(n_runs):
-            if fixed_n_classes is None:
-                labels_a = random_labels(low=0, high=k, size=n_samples)
-            labels_b = random_labels(low=0, high=k, size=n_samples)
-            scores[i, j] = score_func(labels_a, labels_b)
-    return scores
 
+# %%
+# Another function will use the `random_labels` function to create a fixed set
+# of ground truth labels (`labels_a`) distributed in `n_classes` and then score
+# several sets of randomly "predicted" labels (`labels_b`) to assez the
+# variability of a given metric at a given `n_clusters`.
 
-def ami_score(U, V):
-    return metrics.adjusted_mutual_info_score(U, V)
 
+def fixed_classes_uniform_labelings_scores(
+    score_func, n_samples, n_clusters_range, n_classes, n_runs=5
+):
+    scores = np.zeros((len(n_clusters_range), n_runs))
+    labels_a = random_labels(n_samples=n_samples, n_classes=n_classes)
 
-score_funcs = [
-    metrics.adjusted_rand_score,
-    metrics.v_measure_score,
-    ami_score,
-    metrics.mutual_info_score,
-]
+    for i, n_clusters in enumerate(n_clusters_range):
+        for j in range(n_runs):
+            labels_b = random_labels(n_samples=n_samples, n_classes=n_clusters)
+            scores[i, j] = score_func(labels_a, labels_b)
+    return scores
 
-# 2 independent random clusterings with equal cluster number
 
-n_samples = 100
-n_clusters_range = np.linspace(2, n_samples, 10).astype(int)
+# %%
+# In this first example we set the number of clases (true number of clusters) to
+# `n_classes=10`. The number of clusters varies over the values provided by
+# `n_clusters_range`.
 
-plt.figure(1)
+import matplotlib.pyplot as plt
 
+n_samples = 1000
+n_classes = 10
+n_clusters_range = np.linspace(2, 100, 10).astype(int)
 plots = []
 names = []
-for score_func in score_funcs:
-    print(
-        "Computing %s for %d values of n_clusters and n_samples=%d"
-        % (score_func.__name__, len(n_clusters_range), n_samples)
-    )
 
-    t0 = time()
-    scores = uniform_labelings_scores(score_func, n_samples, n_clusters_range)
-    print("done in %0.3fs" % (time() - t0))
+plt.figure(1)
+
+for score_name, score_func in score_funcs:
+
+    scores = fixed_classes_uniform_labelings_scores(
+        score_func, n_samples, n_clusters_range, n_classes=n_classes
+    )
     plots.append(
-        plt.errorbar(n_clusters_range, np.median(scores, axis=1), scores.std(axis=1))[0]
+        plt.errorbar(
+            n_clusters_range, scores.mean(axis=1), scores.std(axis=1), alpha=0.8
+        )[0]
     )
-    names.append(score_func.__name__)
+    names.append(score_name)
 
 plt.title(
-    "Clustering measures for 2 random uniform labelings\nwith equal number of clusters"
+    "Clustering measures for random uniform labeling\n"
+    f"against reference assignment with {n_classes} classes"
 )
 plt.xlabel("Number of clusters (Number of samples is fixed to %d)" % n_samples)
```

Comment:
There are still a few `%`-interpolated strings that should be converted to f-strings in this example (check plot labels and `print` calls).